avalon/nuke/lib.py

```diff
@@ -3,82 +3,283 @@
 import nuke
 import re
 import logging
-import toml
+import six
+from ..vendor import clique
 
 log = logging.getLogger(__name__)
 
 
 @contextlib.contextmanager
 def maintained_selection():
+    """Maintain selection during context
+
+    Example:
+        >>> with maintained_selection():
+        ...     node['selected'].setValue(True)
+        >>> print(node['selected'].value())
+        False
+    """
+    nodes = nuke.allNodes()
     previous_selection = nuke.selectedNodes()
     try:
         yield
     finally:
         # unselect all selection in case there is some
-        current_seletion = nuke.selectedNodes()
-        [n['selected'].setValue(False) for n in current_seletion]
+        [n['selected'].setValue(False) for n in nodes]
```

Comment:
Here you can use the `reset_selection()` that you conveniently wrote below. If the `reset_selection` is really to simple to even be it's own function (which I think it actually is) I'd just change this to:

```python
for node in nuke.selectedNodes():
    node['selected'].setValue(False)
```